homeassistant/components/hue_ble/quality_scale.yaml

```
@@ -0,0 +1,60 @@
+rules:
+  # Bronze
+  action-setup: done
+  appropriate-polling: done
+  brands: done
+  common-modules: done
+  config-flow-test-coverage: done
+  config-flow: done
+  dependency-transparency: done
+  docs-actions: done
+  docs-high-level-description: done
+  docs-installation-instructions: done
+  docs-removal-instructions: done
+  entity-event-setup: done
+  entity-unique-id: done
+  has-entity-name: done
+  runtime-data: done
+  test-before-configure: done
+  test-before-setup: done
+  unique-config-entry: done
+
+  # Silver
+  action-exceptions: done
```

We currently don't catch exceptions raised by for example turning on the light

---

Is it supposed to? The [docs](https://developers.home-assistant.io/docs/core/integration-quality-scale/checklist/) just say actions need to raise an exception, not catch it. If I try to turn the light on while its unplugged I get `Failed to perform the action light.turn_on. Unknown error.` in the web UI.

---

Sorry, rephrase, we need to rethrow the exceptions as HomeAssistantError so we can set the error message

---

Fair enough, I think that's a problem for later, ill mark it as todo for now.